libcodechecker/orm_model.py

```diff
@@ -89,25 +87,31 @@ def __init__(self, run_id, checker_name, attribute, value):
         self.checker_name, self.run_id = checker_name, run_id
 
 
+class FileContent(Base):
+    __tablename__ = 'file_content'
+
+    content_hash = Column(String, primary_key=True)
+    content = Column(Binary)
+
+    def __init__(self, content_hash, content):
+        self.content_hash, self.content = content_hash, content
+
+
 class File(Base):
     __tablename__ = 'files'
 
     id = Column(Integer, autoincrement=True, primary_key=True)
-    run_id = Column(Integer,
-                    ForeignKey('runs.id', deferrable=True,
-                               initially="DEFERRED",
-                               ondelete='CASCADE')
-                    )
     filepath = Column(String)
-    content = Column(Binary)
-    inc_count = Column(Integer)
+    content_hash = Column(String,
+                          ForeignKey('file_content.content_hash',
+                                     deferrable=True,
+                                     initially="DEFERRED", ondelete='CASCADE'))
 
-    def __init__(self, run_id, filepath):
-        self.run_id, self.filepath = run_id, filepath
-        self.inc_count = 0
+    __table_args__ = (UniqueConstraint('filepath', 'content_hash',
+                                       name='_filpath_content_hash_uc'),)
```

Comment:
The name of the `UNIQUE KEY` should be created by SQLAlchemy per the `CC_META` naming convention. There isn't a need to specify a name here.

Make sure to update the migration script to the observed name when you change this!